src/components/FlowPlayer/FlowPlayer.tsx

```diff
@@ -52,42 +52,61 @@ export class FlowPlayer extends React.Component<FlowPlayerProps, FlowPlayerState
 		};
 	}
 
-	componentWillUnmount() {
+	private destroyPlayer() {
 		if (this.state.flowPlayerInstance) {
 			this.state.flowPlayerInstance.destroy();
+			if (this.state.flowPlayerInstance.parentElement) {
+				this.state.flowPlayerInstance.parentElement.innerHTML = '';
+			}
 		}
 	}
 
+	componentWillUnmount() {
+		this.destroyPlayer();
+	}
+
 	shouldComponentUpdate(nextProps: FlowPlayerProps) {
-		if (!this.videoContainerRef.current) {
-			return true;
-		}
+		try {
+			if (!this.videoContainerRef.current) {
+				return true;
+			}
 
-		if (this.state.flowPlayerInstance) {
-			if (nextProps.seekTime !== this.props.seekTime && nextProps.seekTime) {
-				this.state.flowPlayerInstance.currentTime = nextProps.seekTime;
+			const flowPlayerInstance = this.state.flowPlayerInstance;
+			if (flowPlayerInstance) {
+				if (nextProps.seekTime !== this.props.seekTime && nextProps.seekTime) {
+					flowPlayerInstance.currentTime = nextProps.seekTime;
```

Comment:
Hier wordt de state direct gemuteerd

Reply:
mijn IDE dacht dat ook
maar die flowplayerInstance stelt de video player element voor
dus die currentTime is eigenlijk een setter om de huidige tijd in te stellen op het html5 video element dus ik denk niet dat dat een state mutation is.

Reply:
ik heb de code wat aangepast, zodat hij die warning niet meer toont